backend/de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/inout/util/ShipmentScheduleQtyOnHandStorageFactory.java

```diff
@@ -34,30 +32,22 @@
 @Service
 public class ShipmentScheduleQtyOnHandStorageFactory
 {
-	private final StockRepository stockRepository;
+	private final List<IShipmentScheduleQtyOnHandProvider> providers;
 
-	public ShipmentScheduleQtyOnHandStorageFactory(@NonNull final StockRepository stockRepository)
+	public ShipmentScheduleQtyOnHandStorageFactory(@NonNull final List<IShipmentScheduleQtyOnHandProvider> providers)
 	{
-		this.stockRepository = stockRepository;
+		this.providers = providers;
```

Comment:
pls logger.info("Providers: {}", this.providers);

also, i guess it's fine, but be aware that if no provider is found, spring will fail.

Reply:
logged. I don't expect it to ever have less than 1 provider, since a good candidate sits in the same package.